bake/bake.go

```diff
@@ -577,6 +580,11 @@ func (c Config) target(name string, visited map[string]*Target, overrides map[st
 	return tt, nil
 }
 
+type Definition struct {
```

Comment:
It's a bit weird now that we have `Config` and `Definition` (that are both quite meaningless names) and the difference is that one has an array, and the other has maps. Any other name ideas, ways to combine, or I guess introducing this type isn't really needed for the functionality this PR introduces.